### PR TITLE
Add locking to account and submittal of form on backend

### PR DESCRIPTION
**Pull request by bmallred**
Resolves: truetandem/e-QIP-prototype#2380
Resolves: truetandem/e-QIP-prototype#2112

image

### CODE REVIEW

File: api/handlers/save.go [hunk 1]
```diff
@@ -67,6 +82,114 @@ func Hash(w http.ResponseWriter, r *http.Request) {
...
+		log.Warn(logmsg.AccountLocked)
+		EncodeErrJSON(w, err)
+		return
+	}
+
+	EncodeJSON(w, form.Metadata(context, account.ID, account.Locked))
+}
+
+// Submit the application package to the external web service for further processing.
+func Submit(w http.ResponseWriter, r *http.Request) {
```
Comment (floresj) [inline api/handlers/save.go]: Flow looks good to me.

**Comment by bmallred** [comment]
## Mock 1: Redirect to separate view with no additional functionality

image

## Mock 2: Redirect to separate view with only **print/save** functionality

image

The third option is to redirect directly to the **print** section of the form with the menu locked out and allow the user to **print/save** the form.

**Comment by billy22g** [comment]
Option 1 per Sheldon

**Comment by floresj** [comment]
I tried simulating the lock behavior by setting the `locked` column to true in the accounts table and then logging back into the site. The site loaded with none of the values populated but it didn't redirect me to the locked view. Is there something else I need to change to see that?

**Comment by bmallred** [comment]
Hm, I think that should've worked. I personally hardcoded the returned JSON in my testing. Let's sync up when you have a moment and run through it.

**Comment by floresj** [comment]
:tada: 
selection_727

**Pull request merged**